### PR TITLE
feat: add cookie banner

### PR DESCRIPTION
**Pull request by rosejbon**
For this card: https://trello.com/c/CtIuhNWh/131-analytics-and-cookie-page-vv-ee

### CODE REVIEW

**Comment by marxian** [comment]
#### 📝Changed routes:


| [**/cookies**](https://my-best-life-579v8vtvt-neontribe.vercel.app/cookies) |
|:-:|
|<a href="https://my-best-life-579v8vtvt-neontribe.vercel.app/cookies">Screenshot of /cookies</a><br /><sup><a href="https://images.weserv.nl?url=https://iad.microlink.io/Lrbmx4iM86PJ8L6iRR35ZUiqq6fZJ3rNrYGQyr0rGjz1BQhwV5LbWy3-yrQgtK0s7CMmG91iSgVB3qS__6LtnA.png">(view full size)</a> |







Commit 3169bf1fa9c750c192e81bc5e0dfde8ea19d4bfb (https://my-best-life-579v8vtvt-neontribe.vercel.app).

File: pages/_app.tsx [hunk 1]
```diff
@@ -10,6 +11,15 @@ import { NotificationsProvider } from '../src/context/NotificationsContext'
 import { QuizProvider } from '../src/context/QuizContext'
 
 function MyApp({ Component, pageProps }: AppProps): JSX.Element {
+  // Check cookies haven't already been selected and if so don't show banner
+  useEffect(() => {
+    const isHotjarCookiesAccepted = localStorage.getItem(
+      'hotjarCookiesAccepted'
+    )
+    if (isHotjarCookiesAccepted) {
```
Comment (Nikomus) [inline pages/_app.tsx:19]: I think we have to check `if (isHotjarCookiesAccepted === 'false')` here. Local storage only stores strings, and any string with a length greater than 0 is considered `true`,

File: src/Components/CookieBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  const onAccept = () => {
+    // set cookie in local storage
+    localStorage.setItem('hotjarCookiesAccepted', 'true')
+    window.location.reload()
+  }
+
+  const onReject = () => {
+    // set cookie in local storage
+    localStorage.setItem('hotjarCookiesAccepted', 'false')
+    window.location.reload()
```
Comment (Nikomus) [inline src/Components/CookieBanner.tsx:45]: We shouldn't need to refresh if nothing is changing.

File: src/Components/CookieBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  return (
+    <CookieBannerContainer>
+      <p>
+        <b>Can we use cookies to help improve this site?</b>
```
Comment (Nikomus) [inline src/Components/CookieBanner.tsx:50]: We don't have a Lato bold font file so the browser will guess a little bit how to make this. We can add the font if you would like.

File: src/Components/Menu.tsx [hunk 4]
```diff
@@ -48,6 +48,7 @@ export const Menu = ({ open }: MenuProps): JSX.Element => {
       <a href="/">About</a>
       <a href="/helplines">Urgent Help</a>
       <a href="/">Privacy Policy</a>
+      <a href="/cookies">Cookies</a>
```
Comment (Nikomus) [inline src/Components/Menu.tsx:51]: 🙌

File: src/Components/CookieBanner.tsx (hunk 2 above)
Comment (rosejbon) [reply src/Components/CookieBanner.tsx:45]: I need to refresh the page to stop showing the cookie banner after they've made a choice. There is probably a clever way to stop showing it.

File: src/Components/CookieBanner.tsx (hunk 3 above)
Comment (rosejbon) [reply src/Components/CookieBanner.tsx:50]: I will change it to a h3 with some styling that isn't bold.

File: pages/_app.tsx (hunk 1 above)
Comment (rosejbon) [reply pages/_app.tsx:19]: Good spot, I think it needs to be `if (isHotjarCookiesAccepted === 'true')`, as I only want that running when they say yes to cookies.

File: src/Components/CookieBanner.tsx (hunk 3 above)
Comment (Nikomus) [reply src/Components/CookieBanner.tsx]: I think we only guarantee an h1 on the page, can it be an h2?

File: src/Components/CookieBanner.tsx (hunk 2 above)
Comment (rosejbon) [reply src/Components/CookieBanner.tsx:51]: 👍🏻

File: pages/cookies.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                name="hotjarCookiesAccepted"
+                value="false"
+                checked={!cookiesAccepted}
+              />
+              Do not use Hotjar cookies
+            </label>
+          </div>
+          <SubHeading>Strictly necessary cookies</SubHeading>
+          <Text>
+            We use essential cookies to remember your cookie preferences.
```
Comment (Nikomus) [inline pages/cookies.tsx:180]: We don't use a cookie for this, we use local storage 🙂

File: pages/cookies.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+          Cookies are files saved on your phone, tablet or computer when you
+          visit a website.
+        </Text>
+        <Text>
+          We use cookies to store information about how you use this site, such
+          as the pages you visit.
+        </Text>
+        <Heading>Cookie Settings</Heading>
+        <Text>
+          We use 2 types of cookies. You can choose which cookies you are happy
```
Comment (Nikomus) [inline pages/cookies.tsx:126]: Only one type!

File: pages/cookies.tsx (hunk 5 above)
Comment (rosejbon) [reply pages/cookies.tsx:180]: Local storage is considered a "cookie like" technology that we have to tell people about according to ico guidance. (https://ico.org.uk/for-organisations/guide-to-pecr/guidance-on-the-use-of-cookies-and-similar-technologies/what-are-cookies-and-similar-technologies/)

Comment (rosejbon) [reply pages/cookies.tsx:180]: We may have to say we use it for the quiz now I come to think about it

**Review by Nikomus — APPROVED** [review]
Awesome! 🍪

**Pull request merged**